lib/ipmeta.c

```diff
@@ -194,6 +194,44 @@ ipmeta_provider_t **ipmeta_get_all_providers(ipmeta_t *ipmeta)
   return ipmeta->providers;
 }
 
+void ipmeta_record_clear(ipmeta_record_t *record)
+{
+  if (record == NULL) {
+    return;
+  }
+  free(record->region);
+
+  free(record->city);
+
+  free(record->post_code);
+
+  free(record->conn_speed);
+
+  free(record->polygon_ids);
+
+  free(record->asn);
+
+  free(record->sub_name);
+
+  free(record->country);
+
+  free(record->timezone);
+
+  memset(record, 0, 1);
```

Comment:
This writes a single 0 byte to the record structure. 
See `man 3 memset` for usage instructions.
In this case you're going to want something like `memset(record, 0, sizeof(ipmeta_record_t))`.